arangod/IResearch/IResearchAnalyzerFeature.cpp

```diff
@@ -320,15 +327,21 @@ REGISTER_ANALYZER_VPACK(irs::analysis::text_token_stream, text_vpack_builder,
 namespace stem_vpack {
   // FIXME implement proper vpack parsing
   irs::analysis::analyzer::ptr stem_vpack_builder(irs::string_ref const& args) noexcept {
-    return irs::analysis::analyzers::get("stem", irs::text_format::json,
-      arangodb::iresearch::slice<char>(args).toString(),
-      false);
+    auto slice = arangodb::iresearch::slice<char>(args);
+    if (!slice.isNone()) {
+      return irs::analysis::analyzers::get("stem", irs::text_format::json,
+        slice.toString(),
+        false);
+    } 
+    return nullptr;
   }
 
   bool stem_vpack_normalizer(const irs::string_ref& args, std::string& out) noexcept {
     std::string tmp;
-    if (irs::analysis::analyzers::normalize(tmp, "stem", irs::text_format::json,
-      arangodb::iresearch::slice<char>(args).toString(), false)) {
+    auto slice = arangodb::iresearch::slice<char>(args);
+    if (!slice.isNone() && 
+        irs::analysis::analyzers::normalize(tmp, "stem", irs::text_format::json,
+      slice.toString(), false)) {
       auto vpack = VPackParser::fromJson(tmp);
       out.resize(vpack->slice().byteSize());
```

Comment:
While this (resizing and then memcpying) works, it looks a bit complicated.
Why not simply append to the string (which will also reserve enough memory in one go)?
```
out.append(vpack->slice().startAs<char>(), vpack->slice().byteSize());
```
This also makes it impossible to fuck up the memcpy parameters and such.